### ISSUE TITLE
Changing cross entropy loss to hinge loss

### ISSUE THREAD

**Issue by stellaYi** [namisan/mt-dnn#71]
I have applied the model to my own dataset, and trying to explore using hinge loss instead of cross entropy loss for doing binary classification (to determine is the answer is a suitable answer for my question).

However, am not too sure i am doing this correctly.

I have changed this line in model.py:
loss = F.cross_entropy(logits, y)

to:
loss = my_hinge_loss(logits, y.reshape(y.shape[0],1), self.config['margin'])
loss = torch.mean(loss)

########## 
This is my own hinge loss function ( i wrote my own as pytorch embedding hinge loss will result in negative loss):

def my_hinge_loss(logits, labels, margin):
    
    # converts label into -1 and 1
    labels[labels==0] = -1 
    labels = labels.float()

    # while i use relu here, is effective the same as max(0, margin-labels*logits)
    return F.relu(margin-labels*logits, inplace=False)

Question 
Am i correct in passing in logits to compute hinge loss directly?

Referring to 2 results that i have attached (with different margin and learning rate as per indicated):
Any idea why does the ACC fluctuates so much ? Is it because my margin is too wide? is these level of fluctuation normal for hinge lost? 

![Screenshot 2019-07-24 at 4 47 13 PM](https://user-images.githubusercontent.com/42829309/61779315-0a3f5a80-ae33-11e9-9fe4-76364ce81d2a.png)
![Screenshot 2019-07-24 at 4 46 58 PM](https://user-images.githubusercontent.com/42829309/61779316-0ad7f100-ae33-11e9-8594-ac0ef9f53186.png)

**Comment by namisan** [comment]
Please use the officially implemented one or refer it: https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/loss.py#L1013
